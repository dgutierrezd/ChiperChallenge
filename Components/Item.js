import moment from 'moment';
import React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

const Item = ({item, setWebLink}) => {
  return (
    <TouchableOpacity
      style={styles.containerItem}
      onPress={() => setWebLink(item.data?.permalink)}>
      <Text style={styles.fromDate}>
        {moment(moment.unix(item.data?.created_utc)).fromNow()}
      </Text>
      <View style={styles.flexRow}>
        <Image source={{uri: item.data?.thumbnail}} style={styles.imageSize} />
        <View style={styles.textContainer}>
          <Text style={styles.title} numberOfLines={1}>
            {item.data?.title}
          </Text>
          <Text style={styles.author}>{item.data?.author}</Text>
          <View style={styles.bottomContainer}>
            <Text>Score: {item.data?.score}</Text>
            <Text>{item.data?.num_comments} comments</Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  containerItem: {
    marginHorizontal: 5,
    marginVertical: 5,
    flexDirection: 'column',
  },
  fromDate: {textAlign: 'right', fontSize: 10},
  flexRow: {flexDirection: 'row'},
  textContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginLeft: 5,
  },
  title: {fontSize: 20, fontWeight: 'bold', width: 300},
  author: {fontSize: 12},
  bottomContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  imageSize: {width: 50, height: 60},
});

export default Item;
