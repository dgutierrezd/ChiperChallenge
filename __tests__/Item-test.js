import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../Components/Item';
import MockItem from './mock-item.json';

test('renders correctly', () => {
  const tree = renderer.create(<Item item={MockItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});
