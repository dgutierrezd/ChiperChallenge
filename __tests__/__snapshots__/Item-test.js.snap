// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "marginHorizontal": 5,
      "marginVertical": 5,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 10,
        "textAlign": "right",
      }
    }
  >
    11 hours ago
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://a.thumbs.redditmedia.com/isXX5pFIpq5Dp9pPrfYb9kalYmyAS7HOI9H5VLA24u4.jpg",
        }
      }
      style={
        Object {
          "height": 60,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 5,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "width": 300,
          }
        }
      >
        Mullahs who have been harassing women in Iran for 43 years are now being harassed back by everybody
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        chillyfits
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Score: 
          69698
        </Text>
        <Text>
          2034
           comments
        </Text>
      </View>
    </View>
  </View>
</View>
`;
